src/crimson/osd/osd_operations/client_request.cc

```diff
@@ -140,6 +140,11 @@ ClientRequest::interruptible_future<> ClientRequest::with_pg_process_interruptib
 
   DEBUGDPP("{} start", *pgref, *this);
   PG &pg = *pgref;
+
+  DEBUGDPP("{}.{}: entering wait_pg_ready stage",
+	   *pgref, *this, this_instance_id);
+  ihref.enter_stage_sync(client_pp(pg).wait_pg_ready, *this);
```

Comment:
enter_stage would work, but it's strictly less efficient as the interface allows it to block (though it never will).  We're using enter_stage_sync here because the prior stage is PerShardPipeline::create_or_wait_pg (OrderedExclusive) and CommonPGPipeline::WaitPGReady is OrderedConcurrent.  Exiting an OrderedExclusive stage never blocks and entering an OrderedConcurrent stage never blocks.  enter_stage_sync for that transition, therefore, does not need to allow for blocking and is therefore simpler and more efficient.

Reply:
adding a comment